package/MDAnalysis/core/topologyattrs.py

```diff
@@ -3118,9 +3111,8 @@ def _add_bonds(self, values, types=None, guessed=True, order=None):
         if order is None:
             order = itertools.cycle((None,))
 
-        existing = set(self.values)
         for v, t, g, o in zip(values, types, guessed, order):
-            if v not in existing:
+            if v not in self.values:
```

Comment:
This change is likely the culprit - doing a `not in` operation on a list is orders of magnitude slower than on a set.

Example:

```python
In [1]: import random

In [2]: lister = [random.random() for i in range(10000)]

In [3]: setter = set(lister)

In [4]: len(setter)
Out[4]: 10000

In [5]: len(lister)
Out[5]: 10000

In [6]: def notin(a):
   ...:     for v in range(10000):
   ...:         if v not in a:
   ...:             pass
   ...:     return
   ...: 

In [7]: %timeit notin(lister)
734 ms ± 5.85 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [8]: %timeit notin(setter)
129 μs ± 166 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

Reply:
I removed this in the commit below, but note to self that changing `self.values` to a dict might be the easiest way to do this performantly and retain both uniqueness and order.